sentinelsat/sentinel.py

```diff
@@ -318,28 +315,42 @@ def download(self, id, directory_path='.', checksum=False, check_existing=False)
 
         self.logger.info('Downloading %s to %s' % (id, path))
 
-        # Check if the file exists and passes md5 test
-        # The download function will by default continue the download if the file exists but is incomplete
-        if exists(path) and getsize(path) == product_info['size']:
-            if not check_existing or _md5_compare(path, product_info['md5']):
-                self.logger.info('%s was already downloaded.' % path)
-                return product_info
+        if exists(path):
+            # We assume that the product has been downloaded and is complete
+            return product_info
+
+        # Use a temporary file for downloading
+        temp_path = path + '.incomplete'
+
+        skip_download = False
+        if exists(temp_path):
+            if getsize(temp_path) > product_info['size']:
+                remove(temp_path)
```

Comment:
Partial downloads are definitely continued, even with this change. `getsize(temp_path) > product_info['size']` situation deletes the temporary file because its size on disk is larger than it is supposed to be, which should never happen. I should probably log a warning in that situation, though.